src/libstd/ffi/c_str.rs

```diff
@@ -1551,6 +1551,24 @@ impl ops::Index<ops::RangeFull> for CString {
     }
 }
 
+#[stable(feature = "cstr_range_from", since = "1.45.0")]
+impl ops::Index<ops::RangeFrom<usize>> for CStr {
+    type Output = CStr;
+
+    fn index(&self, index: ops::RangeFrom<usize>) -> &CStr {
+        let bytes = self.to_bytes_with_nul();
+        if index.start < bytes.len() {
```

Comment:
Maybe add a comment saying we need our own bounds check to ensure that the NULL terminator is still included.